rpc_jobs/phobos.yml

```diff
@@ -0,0 +1,194 @@
+      }
+      library "rpc-gating@${RPC_GATING_BRANCH}"
+      common.use_node("deploy-phobos"){
+        try {
+          withCredentials([
+              string(
+                credentialsId: "PHOBOS_DEDICATED_ACCOUNT_NUMBER",
+                variable: "RAX_DEDICATED_ACCOUNT_NUMBER"
+              ),
+              usernamePassword(
+                credentialsId: "rpc_jenkins_svc",
+                usernameVariable: "RAX_SSO_USER",
+                passwordVariable: "RAX_SSO_PASSWORD"
+              ),
+              usernamePassword(
+                credentialsId: "phobos_verification_admin",
+                usernameVariable: "PHOBOS_VERIFICATION_ADMIN_USER",
+                passwordVariable: "PHOBOS_VERIFICATION_ADMIN_PASSWORD"
+              )
+              // MAAS_TOKEN is added to env in "Check MaaS Credentials"
+          ]){
+            // do checkout before warning users to ensure branch is valid.
+            stage("Checkout"){
+              dir("rpc-eng-ops"){
+                git branch: env.RPC_ENG_OPS_BRANCH, url: env.RPC_ENG_OPS_REPO, credentialsId: 'rpc-jenkins-svc-github-ssh-key'
+              }
+              // Shell used instead of git function as I want to make all
+              // modifications to the exisiting clone expicit
+              sh """#!/bin/bash -xeu
+                cd /opt/rpc-openstack
+                # log current HEAD
+                git show
+                git status
+
+                pushd openstack-ansible
+                  sudo git stash
+                  sudo git reset --hard
+                popd
+                sudo git fetch --all
+                sudo git stash
+                sudo git reset --hard
+                sudo git checkout "${env.RPC_BRANCH}"
+                sudo git reset --hard origin/${env.RPC_BRANCH}
+                sudo git submodule update --init
+
+                cd /opt/rpc-maas
+                sudo git fetch --all
+                sudo git stash
+                sudo git reset --hard
+                sudo git checkout ${env.RPC_MAAS_BRANCH}
+                sudo git reset --hard origin/${env.RPC_MAAS_BRANCH}
+              """
+            }
+            stage("Check MaaS Credentials"){
+              // MaaS credentials check must be performed on an internal
+              // slave as it may use the internal identity api which
+              // isn't available from phobos.
+              common.internal_slave(){
+                dir("rpc-eng-ops"){
+                  git branch: env.RPC_ENG_OPS_BRANCH, url: env.RPC_ENG_OPS_REPO, credentialsId: 'rpc-jenkins-svc-github-ssh-key'
+                }
+                sh """#!/bin/bash
+                  scl enable python27 ${WORKSPACE}/rpc-eng-ops/phobos-re-automation/maas_creds_check.sh
+                """
+                // Ensure that the MAAS_TOKEN env var is correct
+                // whether the token was obtained from the service account
+                // or supplied via job param.
```

Comment:
Do these comments need to be moved above the actual execution of `maas_creds_check.sh`?

Reply:
No because they related to the line below, where the env var MAAS_TOKEN is re-set.